File: 365. Water and Jug Problem.py
class Solution(object):
    def canMeasureWater(self, x, y, z):
        """
        :type x: int
        :type y: int
        :type z: int
        :rtype: bool
        """
        if x == 0 or y == 0:
            return x == z or y == z
        def gcd(a, b):
            if b == 0:
                return a
            return gcd(b, a%b)
        
        g = gcd(x, y)
        return z <= x + y and z % g == 0
        

File: 365. Water and Jug Problem.go
func canMeasureWater(jug1Capacity int, jug2Capacity int, targetCapacity int) bool {
    var gcd func(int, int) int
	gcd = func(a, b int) int {
		if b == 0 {
			return a
		}
		return gcd(b, a%b)
	}
	return targetCapacity <= jug1Capacity+jug2Capacity && targetCapacity%gcd(jug1Capacity, jug2Capacity) == 0
}